lib/ai/model/generationtool.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package model
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/trace"
+)
+
+type commandGenerationTool struct{}
+
+type commandGenerationToolInput struct {
+	// Command is a unix command to execute.
+	Command string `json:"command"`
+}
+
+func (c *commandGenerationTool) Name() string {
+	return "Command Generation"
+}
+
+func (c *commandGenerationTool) Description() string {
+	// acknowledgement field is used to convince the LLM to return the JSON.
+	// Base on my testing LLM ignores the JSON when the schema has only one field.
+	// Adding additional "pseudo-fields" to the schema makes the LLM return the JSON.
+	return fmt.Sprintf(`Generate a Bash command.
```

Comment:
Should we specify something other than bash here? Maybe we could feed the prompt the OS (if we have that available during an SSH session?)

If not, something more generic would avoid the assumption the user's shell is always bash